extractor/src/main/java/org/schabi/newpipe/extractor/stream/Stream.java

```diff
@@ -2,90 +2,104 @@
+     * @param deliveryMethod the delivery method
+     * @param baseUrl        the base URL of the content if the stream is a DASH or an HLS manifest
      */
-    public Stream(String url, String torrentUrl, MediaFormat format) {
-        this.url = url;
-        this.torrentUrl = torrentUrl;
-        this.format = format.id;
+    public Stream(final String id,
+                  final String content,
+                  final boolean isUrl,
+                  final MediaFormat format,
+                  final DeliveryMethod deliveryMethod,
+                  @Nullable final String baseUrl) {
+        this.id = id;
+        this.content = content;
+        this.isUrl = isUrl;
         this.mediaFormat = format;
+        this.deliveryMethod = deliveryMethod;
+        this.baseUrl = baseUrl;
     }
 
     /**
-     * Reveals whether two streams have the same stats (format and bitrate, for example)
+     * Reveals whether two streams have the same stats (format and bitrate, for example).
      */
-    public boolean equalStats(Stream cmp) {
-        return cmp != null && getFormatId() == cmp.getFormatId();
+    public boolean equalStats(final Stream cmp) {
+        return cmp != null && getFormat().id == cmp.getFormat().id;
     }
 
     /**
-     * Reveals whether two Streams are equal
+     * Reveals whether two streams are equal.
      */
-    public boolean equals(Stream cmp) {
-        return equalStats(cmp) && url.equals(cmp.url);
+    public boolean equals(final Stream cmp) {
+        return equalStats(cmp) && content.equals(cmp.content);
     }
 
     /**
-     * Check if the list already contains one stream with equals stats
+     * Check if the list already contains one stream with equals stats.
      */
-    public static boolean containSimilarStream(Stream stream, List<? extends Stream> streamList) {
+    public static boolean containSimilarStream(final Stream stream,
+                                               final List<? extends Stream> streamList) {
         if (isNullOrEmpty(streamList)) return false;
-        for (Stream cmpStream : streamList) {
+        for (final Stream cmpStream : streamList) {
             if (stream.equalStats(cmpStream)) return true;
         }
         return false;
     }
 
     /**
-     * Gets the url.
-     *
-     * @return the url
+     * Gets the ID for this stream, e.g. itag for YouTube.
+     * @return the id
      */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Gets the URL.
+     *
+     * @return the URL
+     * @deprecated Use {@link #getContent()} instead
+     * */
     public String getUrl() {
-        return url;
+        return isUrl ? content : null;
     }
 
     /**
-     * Gets the torrent url.
+     * Gets the content or URL.
      *
-     * @return the torrent url, example https://webtorrent.io/torrents/big-buck-bunny.torrent
+     * @return the content or URL
      */
-    public String getTorrentUrl() {
-        return torrentUrl;
```

Comment:
I think we are only extracting torrent URLs for PeerTube streams.

Reply:
Only PeerTube provides them and NewPipe can't use them. They are now provided as a separate `Stream` instance with `TORRENT` as `DeliveryMethod`. It also has the same `id` so that clients can now both `Stream`s provide the same file, just with a different `DeliveryMethod`. Or at least that was the idea, but I haven't looked at what @TiA4f8R has changed.

Reply:
Ok. in this case, we can remove that method